### PR TITLE
Refactor most read page margin layout

### PR DESCRIPTION
**Pull request by thekp**
Resolves n/a

**Overall change:**
_We are trying to generalise the index-heading component, it currently accepts a prop of pageType because the most read page does not match the index pages layout, this PR involves refactoring the most read page to re-use the index page layouts.._

**Code changes:**

- _Remove most read page specific margins from index heading container._
- _Remove MarginWrapper from most read container as the section label already adds the necessary spacing through padding._
- _Update snapshots and unit tests_
- _Add a MarginWrapper to most read page to meet the UX requirements of the IDX page._
```
Space around the h1
240-399px: 8px top, 24px bottom
400-599px: 16px top, 24px bottom
600-1007px: 24px top, 32px bottom
1008+px: 24px top, 32px bottom
```

---

- [x] I have assigned myself to this PR and the corresponding issues
- [x] I have added labels to this PR for the relevant pod(s) affected by these changes
- [x] I have assigned this PR to the Simorgh project

**Testing:**

- [x] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [ ] If necessary, I have run the local E2E non-smoke tests relevant to my changes (`CYPRESS_APP_ENV=local CYPRESS_SMOKE=false npm run test:e2e:interactive`)
- [ ] This PR requires manual testing

### CODE REVIEW

File: src/app/pages/MostReadPage/index.jsx [hunk 1]
```diff
@@ -20,81 +21,82 @@ import ChartbeatAnalytics from '#containers/ChartbeatAnalytics';
...
     dir,
     lang,
     mostRead: { header },
   } = useContext(ServiceContext);
 
   const MostReadWrapper = ({ children }) => (
-    <ConstrainedWrapper>
-      <IndexHeading id="content" pageType="mostRead">
+    <>
+      <IndexHeading id="content" pageType="idx">
```
Comment (DenisHdz) [inline src/app/pages/MostReadPage/index.jsx]:
```suggestion
      <IndexHeading id="content">
```

**Review by DenisHdz — COMMENTED** [review]
Thanks for cleaning this up @thekp, it's nice to have the `IndexHeading` back with just one set of paddings. 

I've checked the paddings and they all look fine except one where there is no space:

### **Most Read Page**

**1008+px**
32px under the Heading ✅

most-read-desktop

**600+px**
24px under the Heading ✅

most-read-tablet

**<600px**
24px under the Heading ✅

most-read-mobile

### **Most Read component on FrontPage**

**1008+px**
24px under the section label ✅

most-read-desktop-fp

**600+px**
16px under the section label ❌

most-read-tablet-fp

You can see another section below

section-tablet-example

**<600px**
16px under the section label ✅
most-read-mobile-fp

**Comment by thekp** [comment]
Thanks for the thorough check Denis 💯 . I see the problem lies in min group 3 to max group 3 (600-1007). I don't want to add any space around the most read container, so I updated the section component (used in front page and article) page to add the missing space.

Should we think about refactoring psammead-section-label component to have this spacing also.

**Comment by DenisHdz** [comment]
> Thanks for the thorough check Denis 💯 . I see the problem lies in min group 3 to max group 3 (600-1007). I don't want to add any space around the most read container, so I updated the section component (used in front page and article) page to add the missing space.
> 
> Should we think about refactoring psammead-section-label component to have this spacing also.

Thanks for the changes. I'm also up to not adding space around the `MostRead`container. 

We can look into adding the space to `psammead-section-label` but that would involve removing it from the `story-promo-list` because it is the one adding it between that range. 

I actually think we should look at refactoring the `story-promo-list` because it is making us have that many tweaks everywhere.

**Review by DenisHdz — APPROVED** [review]

**Review by OlgaLyubin — COMMENTED** [review]
Everything looks great 👍 

Just wondering about 240-399px breakpoint. In the PR description, it says that it should have 16px top and 24px bottom. But I see 8px top and 24px bottom

Screenshot 2020-06-17 at 10 15 53

**Comment by thekp** [comment]
@OlgaLyubin Thanks for checking...

The PR description is actually incorrect 🤦 

The intention of this PR is make most read page heading match idx page and the current idx spacing is 8px top and 24px bottom

image

**Review by OlgaLyubin — APPROVED** [review]
Great work:) 
Just needs snapshots conflicts resolved

**Pull request merged**